nixos/modules/services/system/swapspace.nix

```
@@ -9,11 +9,27 @@ let
   cfg = config.services.swapspace;
   inherit (lib)
     types
+    mkIf
     mkOption
     mkPackageOption
     mkEnableOption
     ;
-  configFile = pkgs.writeText "swapspace.conf" (lib.generators.toKeyValue { } cfg.settings);
+  inherit (pkgs)
+    makeWrapper
+    runCommand
+    writeText
+    ;
+  configFile = writeText "swapspace.conf" (lib.generators.toKeyValue { } cfg.settings);
+  userWrapper =
+    runCommand "swapspace"
+      {
+        buildInputs = [ makeWrapper ];
+      }
+      ''
+        mkdir -p $out/bin
+        makeWrapper ${lib.getExe cfg.package} $out/bin/swapspace \
+          --add-flags "-c ${configFile}"
```

```suggestion
        mkdir -p "$out/bin"
        makeWrapper '${lib.getExe cfg.package}' "$out/bin/swapspace" \
          --add-flags "-c '${configFile}'"
```
Let's add some quotes around the variables, as is good practice with bash code. If they contain special characters like spaces, it will still work. Can you please check that the test still runs correctly?

---

yes, I ran the nixosTest.

---

```console
$ nom-build -A nixosTests.swapspace
Finished at 19:22:23 after 9s
/nix/store/3vzc83dqzwqkh0gilq1aszsgx8c415cq-vm-test-run-swapspace
```

---

@Luflosi: fyi, due to how the Nix Store works, none of the variables you quoted will ever contain spaces.